### PR TITLE
Hotfix

### PR DESCRIPTION
**Pull request by drowsy810301**
add has-registered button

### CODE REVIEW

File: contest/contest_info.py [hunk 1]
```diff
@@ -339,10 +339,14 @@ def user_can_register_contest(user, contest):
     has_ownership = has_contest_ownership(user,contest)
     if has_ownership:
         return False
-    is_contestant = Contestant.objects.filter(contest = contest, user = user).exists()
+    return True
+
+def user_has_attended_contest(user, contest):
+    is_contestant = Contestant.objects.filter(contest=contest, user=user).exists()
     if is_contestant:
+        return True
```
Comment (henryyang42) [inline contest/contest_info.py]: ``return Contestant.objects.filter(contest=contest, user=user).exists()`` ?

**Comment by drowsy810301** [comment]
fixed

File: contest/contest_info.py [hunk 2]
```diff
@@ -339,20 +339,23 @@ def user_can_register_contest(user, contest):
...
+def user_has_attended_contest(user, contest):
+    return Contestant.objects.filter(contest=contest, user=user).exists()
+
 '''
 return boolean
 '''
 def can_register(user, contest):
     return (contest_registrable(contest) and user_can_register_contest(user, contest))
 
+def has_attended(user, contest):
```
Comment (nickelcheng) [inline contest/contest_info.py]: What's the difference between ```user_has_attended_contest``` and ```has_attended```?
I think leave one of them is enough.

**Comment by nickelcheng** [comment]
test ok
+1 besides comment

**Comment by drowsy810301** [comment]
comment fixed

**Comment by nickelcheng** [comment]
@bruce3557 please check

File: contest/contest_info.py [hunk 3]
```diff
@@ -339,9 +339,6 @@ def user_can_register_contest(user, contest):
     has_ownership = has_contest_ownership(user,contest)
     if has_ownership:
         return False
```
Comment (nickelcheng) [inline contest/contest_info.py]: By the way, I realize that it's not a good idea to allow public users register contests.
How about add ```if is_public_user(user): return False```

**Comment by drowsy810301** [comment]
comment fixed
+ add ``` if is_public_user(user): return False ```

**Comment by nickelcheng** [comment]
You need ```from contest.public_user import is_public_user``` so that ```is_public_user(user)``` work normally. (it's not work now@@)

**Comment by drowsy810301** [comment]
I forgot.I'm so sorry.

**Comment by drowsy810301** [comment]
comment fixed

File: contest/contest_info.py [hunk 4]
```diff
@@ -61,23 +61,23 @@ def get_running_contests():
     return contests
 
 def get_contestant_list(contest):
-    return Contestant.objects.filter(contest = contest)
+    return Contestant.objects.filter(contest=contest)
 
 def get_total_testcases(problem):
-    testcases = Testcase.objects.filter(problem = problem)
+    testcases = Testcase.objects.filter(problem=problem)
```
Comment (henryyang42) [inline contest/contest_info.py]: +1 for the nice refactor.

**Comment by bruce3557** [comment]
+1, going to merge

**Pull request merged**